### PR TITLE
Milestone 7: Unit Tests

### PR DESCRIPTION
**Pull request by carlylechia**
To complete this milestone, my team and I wrote unit tests with Rspec for the functionality of each and every method in the following classes:

- Book
- Person
- Student
- Teacher
- Classroom
- Rental
- All Decorator pattern classes

The tests were written according to best practices, and they are all passing.

### CODE REVIEW

File: spec/classroom_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      @classroom.add_student(olivier)
+      expect(@classroom.student_array.length).to eql 2
+    end
+
+    it 'Should return the classroom of the student' do
+      olivier = Student.new(25, 'Olivier', parent_permission: true)
+      @classroom.add_student(olivier)
+      expect(olivier.classroom).to eql @classroom
+    end
+  end
```
Comment (julie-ify) [inline spec/classroom_spec.rb]: - [ ] These two tests failed. I've attached the image of the failing tests. Kindly fix them.
chrome_VJcMlEwJUI

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @carlylechia, @RWUBAKWANAYO and @V0pe,

Good job so far!

## Highlights!
- No linter errors ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/classroom_spec.rb (hunk 1 above)
Comment (carlylechia) [reply spec/classroom_spec.rb]: Thank you. I'll try and fix it, although all tests are passing right now on my PC.

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Great work and effort on the project so far and well done making the required changes.
- Great communication with the previous reviewer.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by carlylechia** [comment]
Thank you @MahmoudBakr23

**Pull request merged**